FeedSX/customModals/ReportModal/index.tsx

```diff
@@ -0,0 +1,259 @@
+import {
+  View,
+  Text,
+  Modal,
+  Pressable,
+  TouchableOpacity,
+  TextInput,
+  Image,
+} from 'react-native';
+import React, {useEffect, useState} from 'react';
+import {PostUI} from '../../Models/PostModel';
+import styles from './styles';
+import {useDispatch} from 'react-redux';
+import {getReportTags, postReport} from '../../store/actions/feed';
+import {GetReportTagsRequest, PostReportRequest} from 'likeminds-sdk';
+import {useAppSelector} from '../../store/store';
+import {LMLoader} from '../../components';
+import {
+  COMMENT_REPORT_ENTITY_TYPE,
+  COMMENT_TYPE,
+  POST_REPORT_ENTITY_TYPE,
+  POST_TYPE,
+  REPLY_REPORT_ENTITY_TYPE,
+} from '../../constants/Strings';
+import {showToastMessage} from '../../store/actions/toast';
+
+// interface for post report api request
+interface ReportRequest {
+  entityId: string;
+  entityType: number;
+  reason?: string;
+  tagId: number;
+  uuid: string;
+}
+
+// post report modal props
+interface ReportModalProps {
+  visible: boolean;
+  closeModal: () => void;
+  reportType: string;
+}
+
+type Props = ReportModalProps & PostUI;
+
+const ReportModal: React.FC<Props> = props => {
+  const {visible, closeModal, reportType} = props as ReportModalProps;
+  const {postDetail} = props as PostUI;
+
+  const dispatch = useDispatch();
+  const {Id, uuid} = {...postDetail};
```

Comment:
rename `Id` to `id`

Reply:
we are getting this key from backend like this only.